tests/README.md

```diff
@@ -54,59 +56,13 @@ To write an `istio.io` test, follow these steps:
    Fix the errors, if any, by updating the corresponding command in the `index.md` file and
    then regenerate the snips.
 
-1. Pick an appropriate location under the `tests/` directory for your new test.
-
-1. Create Go boilderplate that will invoke your test bash script using the following pattern:
-
-    ```golang
-    package <your-test-package>
-
-    import (
-        "testing"
-
-        "istio.io/istio/pkg/test/framework"
-
-        "istio.io/istio.io/pkg/test/istioio"
-    )
-
-    func Test<your-test>(t *testing.T) {
-        framework.
-            NewTest(t).
-            Run(istioio.NewBuilder("<your-test-name>").
-                Add(istioio.Script{
-                    Input: istioio.Path("scripts/<your-bash-script>.sh"),
-                }).
-                Defer(istioio.Script{
-                    Input: istioio.Inline{
-                        FileName: "cleanup.sh",
-                        Value: `
-    set +e # ignore cleanup errors
-    source ${REPO_ROOT}/content/en/docs/<your-snips-dir>/snips.sh
-    <your cleanup steps>`,
-                    },
-                }).
-                Build())
-    }
-    ```
-
-    NOTE: This Go boilerplate is a temporary requirement. It will not be needed in the future.
-    See https://docs.google.com/document/d/1r_NoxatNjzPsw0eXr6_9W0rqlkfelt7QfN2y5TF0yTo/edit#.
-
-1. Create your test bash script in the `scripts/` subdirectory.
+1. Create your test bash script `test.sh` next to the `snips.sh` you have just generated. Scripts that have other names will be ignored.
 
 ## Test Bash Script
 
-With the exception of the cleanup steps, your test will consist of a single
-shell scripts that calls the commands in your generated `snips.sh` file.
-
-Your script must include the `snip.sh` file for the document being tested. For example,
-a test for the traffic-shifting task will have the following line in the script:
-
-```sh
-source "${REPO_ROOT}/content/en/docs/tasks/traffic-management/traffic-shifting/snips.sh"
-```
+Your bash script will consist of a series of test steps that call the commands in your generated `snips.sh` file, as well as a series of cleanup steps that should be run after everything is done.
 
-Your test script can then invoke the commands by simply calling snip functions:
+NOTE (for old framework authors): your script doesn't need to include the `snips.sh` or `util/samples.sh` any more. They will be automatically sourced and you can directly call any function defined in them:
```

Comment:
Auto source `snips.sh` is fine, but I don't think we want to do that with `util/samples.sh` because many tests don't need it. In fact, as we write more tests, we may find that there are other common utils that are only needed by a different set of tests (e.g., utils shared by several observability tests) that will be in a similar optional include file (e.g., util/observe.sh). Those kinds of helper functions should not be auto sourced.

Reply:
Sounds reasonable. I'll leave this sourcing up to the test authors. What about `util/verify.sh`? I think sourcing this should be fine.

Reply:
A caveat would be that sourcing needs to be done for both test and cleanup: they are now placed in one single file but are actually run in different shells I believe.

Reply:
There are 4 files that can be auto sourced (since everybody needs them).

1. util/verify.sh
2. util/debug.sh
3. util/helpers.sh
4. snip.sh

Anything else is optional, so it should be explicitly sourced by the test author, instead.

Reply:
Gotcha! I've made the changes accordingly